DevoxxClientMobile/src/main/java/com/devoxx/service/DevoxxService.java

```diff
@@ -1290,4 +1295,18 @@ private Boolean isSavedAccountExpired() {
     private String conferenceUserKey() {
         return getConference().getId() + "_" + encode(username.get());
     }
+
+    private void retrySaveSponsorBadge(SponsorBadge sponsorBadge) {
+        Services.get(ConnectivityService.class).ifPresent(service -> {
+            service.connectedProperty().addListener(new ChangeListener<Boolean>() {
+                @Override
+                public void changed(ObservableValue<? extends Boolean> o, Boolean ov, Boolean nv) {
+                    if (nv) {
+                        saveSponsorBadge(sponsorBadge);
+                    }
+                    service.connectedProperty().removeListener(this);
+                }
+            });
+        });
```

Comment:
You may also want to check if connected is true, and do the task, otherwise add the listener? The failure might not be directly related to lack of connectivity.

Reply:
I don't follow: If `service.isConnected()`, you will add a listener waiting for the connectivity to go off, and only when it is back, try to save the badges?
Or did you mean `if (! service.isConnected())`?

What I had in mind was something like:

```
if (service.isConnected()) {
    saveSponsorBadge(sponsorBadge);
} else {
    service.connectedProperty().addListener((o, ov, nv) -> {
             ...
    });
}
```

Reply:
Sorry, I made a typo and the commit message reflected what I did in the code. I have fixed it now. There is no need for the true branch as the method is called from `saveSponsorBadge()` method.